### ISSUE TITLE
big regression bug  when you copy text with style and there is more than one style in the text, everything gets messy

### ISSUE THREAD

**Issue by jonasbernardo** [singerdmx/flutter-quill#2041]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Flutter Quill version

9.5.23

### Steps to reproduce

write anything, put a background color

skip a line write something and put a color in the text

copy and paste everything written

### Expected results

text is copied with the same formatting

### Actual results

the text style is all messed up
https://github.com/user-attachments/assets/04b29150-9540-4ec7-9ee7-e42f25181d58

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```

</details>

### Additional Context

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

https://github.com/user-attachments/assets/7558f8fe-a92b-431d-87f0-6e30feefc6b1

</details>

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

**Comment by jonasbernardo** [comment]
From what I remember this problem had been fixed and now it happened again, from what I remember it happens because the buttons on the toolbar remain pressed and it ends up messing up what you pasted

**Comment by jonasbernardo** [comment]
in my changelog this warning was removed 

 quill_controller.dart line 268
          // pasted text should not use toggledStyle
          (data is! String || data.length < 2);

**Comment by CatHood0** [comment]
@EchoEllet are you familiar with copy and pasting feature?

**Comment by AtlasAutocode** [comment]
It appears that the pasting is merging the incoming styles with the existing styles at the point of insertion.
To see this: at the bottom, add some bold text and paste the colored text - result is **bold** colored text!

I seem to recollect that pasting from the same editor bypasses the clipboard and handles the paste by special processing. So, I would guess that is where the problem is - probably in the QuillController clipboardPaste handler.
I think the code handling paste plain text has a special call to recognize when the clipboard text matches the plain text for the last copy action.

**Comment by EchoEllet** [comment]
> @EchoEllet are you familiar with copy and pasting feature? 

I'm familiar with the code regarding the copy and paste the HTML and markdown. It was initially in the editor, then moved to the controller to support paste from a toolbar button.

**Comment by CatHood0** [comment]
> probably in the QuillController clipboardPaste handler

I'll take a look this

**Comment by CatHood0** [comment]
I don't know if I'm wrong, but it may be related to the problem.

According to what I see, it should have a `Delta` such that:

```dart
[
 {"insert":"\n"},
 {"insert":"Text","attributes":{"color":"<any_color>"}},
 {"insert":"\n\n"},
 {"insert":"Jonas\n","attributes": {"color":"<any_color>","background":"<any_color>"}},
 {"insert":"\n"}
]
```

_If you ask why "Jonas" also contains a new line, watch the video carefully and come back to this part. You will notice that in the new line that opens, there are still the `color` and `background` attributes_.

So, with this in mind, we can start with the following. As far as I know, for mobile devices, the features for Rich Text Pasting are not available, that is, you are copying the plain text from the editor directly to the Clipboard.

## Where is the problem with this?

The way I see it, that's how it should be working, because the OP is not taking this aspect into account.

## Why are you copying the attributes of the last insertion then?

This is because yes, the OP copies all the text (not new lines), but it also opens a new line which is very likely containing the `background` attribute, and when pasting the text, it gives that feeling that The text is formatted only when in reality it is far from being that way.

That is, it goes from having a Delta like the previous one to having a `Delta` like this:

```dart
[
 {"insert":"Text","attributes":{"color":"<any_color>"}},
 {"insert":"\n\n"},
 {"insert":"Jonas\nText\n\nJonas","attributes": {"color":"<any_color>","background":"<any_color>"}},
 {"insert":"\n"},
]
```

In conclusion, it could be more a matter of the OP being confused and actually doing the job as it should, but not realizing that the `color` and `background` attributes are still applied, it may seem that something is wrong.

**Comment by singerdmx** [comment]
@jonasbernardo May I close this?

**Comment by jonasbernardo** [comment]
the problem has not been resolved, and according to the code, it says the following in quill_controller.dart 
  /// Clipboard caches last copy to allow paste with styles. Static to allow paste between multiple instances of editor.
This works, but only if the text has a type of style

**Comment by jonasbernardo** [comment]
and even disabling the colors in the edit bar, the formatting appears messy

**Comment by jonasbernardo** [comment]
@singerdmx 
@EchoEllet 
@CatHood0

**Comment by jonasbernardo** [comment]
There are several unexpected behaviors, the person just wants to copy the text without formatting or with formatting and neither of them works

**Comment by jonasbernardo** [comment]
https://github.com/user-attachments/assets/29ac62ff-60ab-47c0-8dfa-63ef81c835c0
